utils/wasm-instrument/src/lib.rs

```diff
@@ -213,17 +204,27 @@ pub fn inject<R: Rules>(
     ];
 
     // determine cost for successful execution
+    let mut block_of_code = false;
+
     let cost_blocks = match elements
         .iter()
-        .take(7)
-        // block with update instructions
-        .chain(elements.iter().skip(10).take(7))
+        .filter(|instruction| match instruction {
+            Instruction::If(_) => {
+                block_of_code = true;
+                true
```

Comment:
should we include if cost?

Reply:
> Yes, see how the previous implementation works with `.chain()`

But it may be initial issue xd

Reply:
@gshep

Reply:
@breathx when `Instruction::If` is executed, a jump occurs either to an unreachable part of the code, or to the instruction after `Instruction::End`